gadgets/trace_capabilities/test/trace_capabilities_test.go

```diff
@@ -0,0 +1,150 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	gadgettesting "github.com/inspektor-gadget/inspektor-gadget/gadgets/testing"
+	igtesting "github.com/inspektor-gadget/inspektor-gadget/pkg/testing"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/containers"
+	igrunner "github.com/inspektor-gadget/inspektor-gadget/pkg/testing/ig"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/match"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/utils"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+type traceCapabilitiesEvent struct {
+	eventtypes.CommonData
+
+	MountNsID     uint64 `json:"mntns_id"`
+	Timestamp     string `json:"timestamp"`
+	CurrentUserNs uint64 `json:"current_user_ns"`
+	TargetUserNs  uint64 `json:"target_user_ns"`
+	CapEffective  string `json:"cap_effective"`
+	Pid           uint32 `json:"pid"`
+	Cap           string `json:"cap"`
+	Tgid          uint32 `json:"tgid"`
+	Uid           uint32 `json:"uid"`
+	Gid           uint32 `json:"gid"`
+	Audit         uint32 `json:"audit"`
+	Insetid       uint32 `json:"insetid"`
+	Syscall       string `json:"syscall"`
+	Task          string `json:"task"`
+	Kstack        string `json:"kstack"`
+	Capable       bool   `json:"capable"`
+}
+
+func TestTraceCapabilities(t *testing.T) {
+	gadgettesting.RequireEnvironmentVariables(t)
+	utils.InitTest(t)
+
+	containerFactory, err := containers.NewContainerFactory(utils.Runtime)
+	require.NoError(t, err, "new container factory")
+	containerName := "test-trace-capabilities"
+	containerImage := "docker.io/library/busybox:latest"
+
+	var ns string
+	containerOpts := []containers.ContainerOption{
+		containers.WithContainerImage(containerImage),
+		containers.WithStartAndStop(),
+	}
+
+	if utils.CurrentTestComponent == utils.KubectlGadgetTestComponent {
+		ns = utils.GenerateTestNamespaceName(t, "test-trace-capabilities")
+		containerOpts = append(containerOpts, containers.WithContainerNamespace(ns))
+	}
+
+	testContainer := containerFactory.NewContainer(containerName, "nice -n -20 echo", containerOpts...)
+
+	var runnerOpts []igrunner.Option
+	var testingOpts []igtesting.Option
+	commonDataOpts := []utils.CommonDataOption{
+		utils.WithContainerImageName(containerImage),
+		// TODO: We need to start the container after the tracer, so we don't
+		// have the container ID available here. It could be possible to split
+		// the logic to have a container create + container start to be able to
+		// get the ID before starting it.
```

Comment:
I think this test can be implemented similar to https://github.com/inspektor-gadget/inspektor-gadget/blob/main/gadgets/trace_open/test/trace_open_test.go. (Running the container before the tracer and the test command within a loop).